src/app/shared/components/xterm-terminal/component.ts

```diff
@@ -0,0 +1,215 @@
+})
+export class XTermTerminal implements OnInit, AfterViewInit {
+  readonly DELAY_TIMEOUT = 100;
+  projectId: string;
+  clusterId: string;
+  cluster: Cluster;
+  terminal: Terminal;
+  isConnectionLost: boolean;
+
+  // Note: Hide button when terminal open as new tab
+  @Input() showCloseButtonOnStatusToolbar = true;
+  @Output() terminalDisposed = new EventEmitter<void>();
+  @ViewChild('terminal', {static: true}) terminalRef: ElementRef;
+
+  private readonly _unsubscribe = new Subject<void>();
+  private _resizeEventlistenerFn: () => void;
+
+  constructor(
+    private readonly _activatedRoute: ActivatedRoute,
+    private readonly _clusterService: ClusterService,
+    private readonly _notificationService: NotificationService,
+    private readonly _renderer: Renderer2,
+    private readonly websocketService: WebsocketService
+  ) {}
+
+  ngOnInit(): void {
+    this.initSubscriptions();
+  }
+
+  initSubscriptions() {
+    this.projectId = this._activatedRoute.snapshot.paramMap.get(PathParam.ProjectID);
+    this.clusterId = this._activatedRoute.snapshot.paramMap.get(PathParam.ClusterID);
+
+    // Note:
+    // Get Cluster Info inorder to show cluster name in shell
+    if (this.projectId && this.clusterId) {
+      this._clusterService
+        .cluster(this.projectId, this.clusterId)
+        .pipe(take(1))
+        .subscribe({
+          next: (cluster: Cluster) => {
+            this.cluster = cluster;
+            this.initTerminal();
+          },
+          error: _ => {},
+        });
+    }
+
+    // Subscription bidirectional communication
+    this.websocketService.messages$.pipe(takeUntil(this._unsubscribe)).subscribe((data: ITerminalFrame) => {
+      this.handleWebSocketConnectionMessages(data);
+    });
+
+    // Subscription connection success
+    this.websocketService
+      .getWebSocketOnConnectObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this._notificationService.success('Successfully connected to cloud shell terminal');
+        this.isConnectionLost = false;
+      });
+
+    // Subscription connection lost
+    this.websocketService
+      .getWebSocketOnCloseObservable$()
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(() => {
+        this.isConnectionLost = true;
+      });
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+    this.websocketService.close();
+
+    // Clean 'resize' Event listener subscription
+    this.removeEventListener();
+  }
+
+  ngAfterViewInit(): void {
+    this.connectToWebSocketConnection();
+  }
+
+  initTerminal(): void {
+    if (this.terminal) {
+      this.terminal.dispose();
+    }
+
+    this.terminal = new Terminal({
+      fontSize: 14,
+      fontFamily: 'Consolas, "Courier New", monospace',
+      bellStyle: 'sound',
+      cursorBlink: true,
+      theme: {
+        foreground: '#f8f8f8',
+        background: '#2b3035',
+      },
+    });
```

Comment:
Export config to readonly variable.

Reply:
const are created at top of the file to have all Terminal related configurations.